modules/seqexec/server/src/test/scala/seqexec/server/tcs/TestTcsEpics.scala

```diff
@@ -0,0 +1,1067 @@
+package seqexec.server.tcs
+import cats.{Applicative, Eq}
+import cats.effect.{Sync, Timer}
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import edu.gemini.seqexec.server.tcs.{BinaryOnOff, BinaryYesNo}
+import monocle.{Getter, Lens}
+import monocle.macros.Lenses
+import seqexec.model.enum.ApplyCommandResult
+import seqexec.server.TestEpicsCommand._
+import seqexec.server.EpicsCommand
+import seqexec.server.tcs.TcsEpics._
+import squants.{Angle, Time}
+import squants.space.AngleConversions._
+
+class TestTcsEpics[F[_]: Sync](state: Ref[F, TestTcsEpics.State], out: Ref[F, List[TestTcsEpics.TestTcsEvent]])
+  extends TcsEpics[F] {
+  import TestTcsEpics._
```

Comment:
The test implementation of `TestEpics`. It implements all attributes and certain commands. The implemented commands modify the attributes. `outputF` can be used to retrieve the list of all the TCS commands executed.